### PR TITLE
[tests-only][full-ci] Add tests to list space files shared with user who is also member of group 

### PR DESCRIPTION
**Pull request by nabim777**
## Description

In this PR following scenarios are added related to sharingNG:
```
-   user who is also a member of group lists file shared with them from project-space
-   user who is also a member of group lists folder shared with them from project-space
```

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/8318

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 1]
```diff
@@ -1922,3 +1918,452 @@ Feature: listing sharedWithMe when auto-sync is disabled
         }
       }
       """
+
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:1921]:
```suggestion


```

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 2]
```diff
@@ -1922,3 +1918,452 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+                  "type": "string",
+                  "pattern": "^%share_id_pattern%$"
+                },
+                "name": {
+                  "type": "string",
+                  "enum": ["folderToShare"]
+                },
+                "parentReference": {
+                  "type": "object",
+                  "required": ["driveId", "driveType", "id"],
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2231]:
```suggestion
                  "required": [
                  "driveId", 
                  "driveType", 
                  "id"
                 ],
```

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 3]
```diff
@@ -1922,3 +1918,452 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+                      "type": "string",
+                      "pattern": "^%file_id_pattern%$"
+                    },
+                    "name": {
+                      "type": "string",
+                      "enum": ["folderToShare"]
+                    },
+                    "parentReference": {
+                      "type": "object",
+                      "required": ["driveId", "driveType"],
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2273]:
```suggestion
                      "required": [
                      "driveId", 
                      "driveType"
                     ],
```

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 4]
```diff
@@ -1922,3 +1918,633 @@ Feature: listing sharedWithMe when auto-sync is disabled
         }
       }
       """
+
+    
+  Scenario: user who is also a member of group lists file shared with them from project-space when auto-sync is disabled
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:1923]:
```suggestion
  Scenario: user who is also a member of group lists file shared with them from project-space
```
The feature file name and description at the top of the feature file are clear about the scenarios covered in this file. So, let's not add the text in every description.

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 5]
```diff
@@ -1922,3 +1918,633 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+                        "enum": ["text/plain"]
+                      }
+                    }
+                  },
+                  "id": {
+                    "type": "string",
+                    "pattern": "^%share_id_pattern%$"
+                  },
+                  "name": {
+                    "type": "string",
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:1998]: Try to minimize the line numbers of a schema by removing unnecessary checks. (The required key is sufficient for some checks) Refer to other added scenarios.

Comment (nabim777) [reply tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature]: I have try to minimized can you check if that is ok or not  @grgprarup

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 6]
```diff
@@ -2790,3 +2790,285 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+      | Brian    | grp1      |
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "projectSpace" with the default quota using the Graph API
+    And user "Alice" has uploaded a file inside space "projectSpace" with content "to share" to "textfile.txt"
+    And user "Alice" has sent the following share invitation:
+      | resource        | textfile.txt |
+      | space           | projectSpace |
+      | sharee          | grp1         |
+      | shareType       | group        |
+      | permissionsRole | File Editor  |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2809]:
```suggestion
      | permissionsRole | Viewer  |
```

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 7]
```diff
@@ -2790,3 +2790,285 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+      | space           | projectSpace  |
+      | sharee          | grp1          |
+      | shareType       | group         |
+      | permissionsRole | Viewer        |
+    And user "Alice" has sent the following share invitation:
+      | resource        | folderToShare |
+      | space           | projectSpace  |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Editor        |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2957]:
```suggestion
      | permissionsRole | Viewer        |
```

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature (hunk 6 above)
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2809]: let's share with different role with different sharee.

Comment (grgprarup) [reply tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2809]: > let's share with different role with different sharee.

Does that make any sense in this scenario?

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2809]: there is a bug report on losing permissions of shares from same file (https://github.com/owncloud/ocis/issues/8464). so we can check that different permissions are in intact.

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 8]
```diff
@@ -2790,3 +2790,285 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+                    "parentReference",
+                    "permissions",
+                    "size"
+                  ],
+                  "properties": {
+                    "permissions": {
+                      "type": "array",
+                      "minItems": 2,
+                      "maxItems": 2,
+                      "items": {
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2868]: please, also add `uniqueItems`
```suggestion
                      "maxItems": 2,
                      "uniqueItems": true,
                      "items": {
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature (hunk 8 above)
Comment (nabim777) [reply tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature]: Okay

**Review by grgprarup — APPROVED** [review]

**Review by saw-jan — COMMENTED** [review]
@nabim777 Please, rebase the PR

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 9]
```diff
@@ -2814,3 +2814,352 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+      | permissionsRole | Viewer       |
+    When user "Brian" lists the shares shared with him using the Graph API
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should match
+      """
+      {
+        "type": "object",
+        "required": [
+          "value"
+        ],
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:2848]:
```suggestion
        "required": ["value"],
```

File: tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature [hunk 10]
```diff
@@ -2814,3 +2814,352 @@ Feature: listing sharedWithMe when auto-sync is disabled
...
+                              },
+                              "roles": {
+                                "type": "array",
+                                "minItems": 1,
+                                "maxItems": 1,
+                                "items":
+                                  {
+                                    "type": "string",
+                                    "pattern": "^%role_id_pattern%$"
+                                  }
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedWithMeSyncDisabled.feature:3106]:
```suggestion
                                "items": {
                                  "type": "string",
                                  "pattern": "^%role_id_pattern%$"
                                }
```

**Review by saw-jan — COMMENTED** [review]
we can reduce some lines here

**Review by saw-jan — APPROVED** [review]

**Pull request merged**